### PR TITLE
feat(pattern/inputs): add left icon support

### PR DESCRIPTION
**Pull request by gael-boyenval**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/adeo/design-system--styleguide/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://design-system-adeo.atlassian.net/browse/MZC-51

## What is the new behavior?

adding left icon styles and docs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: src/pages/Components/TextInput/index.md [hunk 1]
```diff
@@ -7,52 +7,53 @@ order: 2
...
-## Do's and don'ts
+## Using a left Icon :
 
-<hintitem>
-You can cumulate différents states like that :
-</hintitem>
-<preview path="src/pages/Components/TextInput/previews/input-invalid-event"></preview>
+Use a left icon to inform about a specific information type or format you expect the user to provide.
+
+<preview nude="true" path="src/pages/Components/TextInput/previews/left-icon-input"></preview>
```
Comment (caqueste) [inline src/pages/Components/TextInput/index.md]: Specs-input
Can we change the height value of the inputs ? They should have a height of 48px.
For the alignement of the placeholder, It must take the lower-case of the placeholder and center it with the input height.

Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: https://design-system-adeo.atlassian.net/browse/MZC-90

Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: Can you review just the feature described by the ticket, I will fix the input layout with the related ticket

File: src/pages/Components/TextInput/index.md [hunk 2]
```diff
@@ -7,52 +7,53 @@ order: 2
...
 
 ### Invalid
 
 When the entry is invalid against the criteria determined upstream.
 
-<preview path="src/pages/Components/TextInput/previews/input-invalid-state"></preview>
+<preview nude="true" path="src/pages/Components/TextInput/previews/input-invalid-state"></preview>
 
-## Do's and don'ts
+## Using a left Icon :
```
Comment (caqueste) [inline src/pages/Components/TextInput/index.md]: **I propose :**
Inputs with icon :

File: src/pages/Components/TextInput/index.md [hunk 3]
```diff
@@ -7,52 +7,53 @@ order: 2
...
 
-<hintitem>
-You can cumulate différents states like that :
-</hintitem>
-<preview path="src/pages/Components/TextInput/previews/input-invalid-event"></preview>
+Use a left icon to inform about a specific information type or format you expect the user to provide.
```
Comment (caqueste) [inline src/pages/Components/TextInput/index.md]: i don't understand the term "format" ? Can we not just speak about action or feature?

File: src/pages/Components/TextInput/index.md (hunk 1 above)
Comment (caqueste) [inline src/pages/Components/TextInput/index.md]: You have to put the icon in 32px for the hierarchy and the icon visibility. We always have complexer icons to the left than the right.
Moreover, the right icon has 24px, so if the left icon has 24px too, they cancel themselves.

File: src/pages/Components/TextInput/index.md (hunk 3 above)
Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: formatting would be better

File: src/pages/Components/TextInput/index.md (hunk 1 above)
Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: 32px is enormous, I think we should use the 16px icon on the validation instead.

Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: I think what's important at the validation step, is the validation, plus the color make the hierarchy for you. even tiny red will always stand out. 

But I would definitely use the 16px icon on the validation. 

what do you think ?

Comment (caqueste) [reply src/pages/Components/TextInput/index.md]: I did some tests in sketch.
Ok to keep the left icon in 24px, and put the right icon in 16px but we must use the icon *Navigation_Notification_Check_16px* for the valid state and the *Navigation_Display_Close_16px* for the error state.

Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: like props in the mineral line 🤩

Comment (gael-boyenval) [reply src/pages/Components/TextInput/index.md]: done

Comment (caqueste) [reply src/pages/Components/TextInput/index.md]: top

**Pull request merged**